cmd/probe/eventhandler.go

```diff
@@ -54,7 +54,29 @@ func (pe *ProbeEvent) addDiskEvent(msg controller.EventMessage) {
 		}
 		glog.Info("processed data for ", diskDetails.ProbeIdentifiers.Uuid)
 		oldDr := pe.Controller.GetExistingResource(diskList, diskDetails.ProbeIdentifiers.Uuid)
-		pe.Controller.PushDiskResource(oldDr, diskDetails)
+		// if old DiskCR doesn't exist and parition is found, it is ignored since we don't need info
+		// of partition if disk as a whole is ignored
+		if oldDr == nil && len(diskDetails.PartitionData) != 0 {
+			glog.Info("Skipping partition of already excluded disk ", diskDetails.ProbeIdentifiers.Uuid)
+			continue
+		} else
+		// if diskCR is already present, and udev event is generated for partition, append the partition info
+		// to the diskCR
+		if oldDr != nil && len(diskDetails.PartitionData) != 0 {
+			newDrCopy := oldDr.DeepCopy()
+			glog.Info("Appending partition data to ", diskDetails.ProbeIdentifiers.Uuid)
+			newDrCopy.Spec.PartitionDetails = append(newDrCopy.Spec.PartitionDetails, diskDetails.ToPartition()...)
+			pe.Controller.UpdateDiskCR(newDrCopy)
+		} else {
+			pe.Controller.PushDiskResource(oldDr, diskDetails)
+		}
+		/// update the list of DiskCRs
+		diskList, err = pe.Controller.ListDiskResource()
```

Comment:
I hope diskList is not causing any memory leak, as we have to call ListDiskResource many a time for each update/push to etcd.

Reply:
diskList is not causing a memory leak. Only pointers that we use in C Go, needs to be explicitly deallocated. The memory that we use here will be automatically garbage collected.